src/fidesops/service/connectors/saas_connector.py

```diff
@@ -104,36 +117,85 @@ def create_client(self) -> AuthenticatedClient:
         return AuthenticatedClient(uri, self.configuration)
 
     def retrieve_data(
-        self, node: TraversalNode, policy: Policy, input_data: Dict[str, List[Any]]
+        self,
+        node: TraversalNode,
+        policy: Policy,
+        privacy_request: PrivacyRequest,
+        input_data: Dict[str, List[Any]],
     ) -> List[Row]:
         """Retrieve data from SaaS APIs"""
 
         # get the corresponding read request for the given collection
-        collection_name = node.address.collection
-        read_request = self.endpoints[collection_name].requests["read"]
+        collection_name: str = node.address.collection
+        read_request: SaaSRequest = self.endpoints[collection_name].requests["read"]
 
-        query_config = self.query_config(node)
-        prepared_requests = query_config.generate_requests(input_data, policy)
+        query_config: SaaSQueryConfig = self.query_config(node)
+        prepared_requests = query_config.generate_query(input_data, policy)
 
         rows: List[Row] = []
         for prepared_request in prepared_requests:
-            response = self.client().send(prepared_request)
-
-            # process response
-            if read_request.data_path:
-                processed_response = pydash.get(response.json(), read_request.data_path)
+            response: Response = self.client().get(prepared_request)
+
+            if read_request.postprocessors is None:
+                rows.extend(response.json())
+                continue
+            data_to_be_processed: Any = self.post_process(
+                node.address,
+                privacy_request.get_cached_identity_data(),
+                read_request.postprocessors,
+                response,
+            )
+            if isinstance(data_to_be_processed, list):
+                filter_only_dict = [
+                    item for item in data_to_be_processed if isinstance(item, dict)
+                ]
+                if len(filter_only_dict) < len(data_to_be_processed):
+                    logger.warning(
+                        "Some data could not be added due to unexpected format"
+                    )
+                rows.extend(data_to_be_processed)
```

Comment:
After testing this, instead of warning, I think we should raise an Exception.  This still allows an array to be added, which would break with a cryptic message  `ERROR:root:0` that's hard to debug. A test would be nice here, I faked it by forcing `data_to_be_processed = [[{"data": "bad"}]]`